cmd/zpool/zpool_main.c

```diff
@@ -2163,19 +2173,84 @@ show_import(nvlist_t *config)
 	}
 }
 
+typedef long long unsigned int llu;
+
+static void
+extract_timestamp_and_txg(nvlist_t *config, uint64_t *txg,
+    uint64_t *timestamp)
+{
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_TIMESTAMP,
+	    timestamp) == 0);
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_FOUND_TXG,
+	    txg) == 0);
+}
+
+static int
+check_pool_for_activity(libzfs_handle_t *hdl, nvlist_t *config)
+{
+	nvlist_t *nvl = NULL;
+	uint64_t original_txg, original_timestamp;
+	int rc = 0;
+	char *poolname;
+	int seconds_between_checks = 1;
+	int times_to_check = 10;
```

Comment:
Be aware that 1-second heartbeat is too small for covering real-world conditions where under heavy loads, I/Os can take longer than 1-second. Also, temporary, transient conditions like HBA or drive resets are on the order of 10 to 15 seconds.

As such, systems designers may wish to change the seconds_between_checks to reduce wearout on limited-endurance media or increase times_to_check to account for transients in their system. Can we make these globally tunable via params?